src/ring/middleware/json.clj

```diff
@@ -100,12 +102,18 @@
        (handler request respond raise)
        (respond malformed-response)))))
 
+(defn- generate-input-stream
+  [body options]
+  (ring-io/piped-input-stream
+   (fn [out] (json/generate-stream body (io/writer out) options))))
+
 (defn json-response
   "Converts responses with a map or a vector for a body into a JSON response.
   See: wrap-json-response."
-  [response options]
+  [response {:keys [stream?] :as options}]
```

Comment:
Destructuing in the arg list affects the documentation generator, so it either needs to be overridden with metadata, or just use `(:stream? options)` instead.